File: Algorithms/Python/searching/binary_search.py
from __future__ import annotations


def binary_search(arr: list[int], target: int) -> int:
    """
    Binary search algorithm to find the index of the target value in the array.
    If the target value is not in the array, return -1.
    """
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == target:
            return mid
        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1


if __name__ == "__main__":
    numbers = list(range(1, 15))
    print(binary_search(numbers, 2))  # 2

File: Algorithms/Javascript/searching/binarySearch.js

const binarySearch = (arr, target) => {
    let left = 0;
    let right = arr.length - 1;

    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        if (arr[mid] === target) {
            return mid;
        }
        if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1;
};

const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

index = binarySearch(arr, 5);
if (index !== -1) {
    console.log(`Element found at index ${index}`);
}
else {
    console.log('Element not found');
}
